contracts/src/StabilityPool.sol

```diff
@@ -408,7 +409,7 @@ contract StabilityPool is LiquityBase, IStabilityPool, IStabilityPoolEvents {
         uint256 yieldPerUnitStaked = yieldNumerator / totalBoldDepositsCached;
         lastYieldError = yieldNumerator - yieldPerUnitStaked * totalBoldDepositsCached;
 
-        uint256 marginalYieldGain = yieldPerUnitStaked * P;
+        uint256 marginalYieldGain = yieldPerUnitStaked * (P - 1);
```

Comment:
I don't fully get the use of `P - 1` rather than P. As far as I can see:

- The P error correction increases P, but
- P error correction itself has rounding error (so, smaller than it should be)

Why do we then use P -1? Doesn't that make it even smaller than it should be?

Reply:
The rationale is that we end up using `x * P_n / P_m`, with `n > m` (and being `P_n` the one in the line above. 
The fraction `P_n / P_m` represents the ratio of total deposits between those two snapshots (`m` and `n`). 
We want the ratio between `P`’s to always be <= than the ratio between deposits. So that rewards shared are less or equal than should be, but not greater.
As both `P_n` and `P_m` are inaccurate, it may happen than the error (truncating) in `P_m` is bigger than in `P_n`. That would make the fraction to increase.
Let’s put an example, assuming `DECIMAL_PRECISION = 100`. Let’s imagine that the total deposits are 4999 and 2001 for `m` and `n` snapshots, and that theoretical values of `P` would be 49.99 (truncated to 49) and 20.01 (truncated to 20). Then we have `P_n / P_m = 20 / 49 > 2001 / 4999 = deposit_n / deposit_m`. That’s what we wan to avoid, so we use `P_n = 19` instead of  20. Of course in this example this makes the precision of `P_n / P_m` very bad, but this is an edge case and mostly due to the low precision (only 2 decimals). With 18 decimals, subtracting 1 wouldn’t be such big deal.

Reply:
OK, yes I see. So this correction should always make the error in `P_n` bigger than the error in `P_m`. Cool, it makes sense.